src/zulip.rs

```diff
@@ -302,10 +319,77 @@ async fn handle_command<'a>(
                 .map_err(|e| format_err!("Failed to await at this time: {e:?}")),
             StreamCommand::PingGoals(args) => ping_goals_cmd(ctx, gh_id, message_data, &args).await,
             StreamCommand::DocsUpdate => trigger_docs_update(message_data, &ctx.zulip),
+            StreamCommand::Backport(args) => {
+                accept_decline_backport(message_data, &ctx.octocrab, &ctx.zulip, &args).await
+            }
         }
     }
 }
 
+// TODO: shorter variant of this command (f.e. `backport accept` or even `accept`) that infers everything from the Message payload
+async fn accept_decline_backport(
+    message_data: &Message,
+    octo_client: &Octocrab,
+    zulip_client: &ZulipClient,
+    args_data: &BackportArgs,
+) -> anyhow::Result<Option<String>> {
+    let message = message_data.clone();
+    let args = args_data.clone();
+    let stream_id = message.stream_id.unwrap();
+    let subject = message.subject.unwrap();
+    let verb = args.verb.to_lowercase();
+    let octo_client = octo_client.clone();
+
+    // Repository owner and name are hardcoded
+    // This command is only used in this repository
+    let repo_owner = "rust-lang";
+    let repo_name = "rust";
+
+    // validate command parameters
+    if !contains_any(&[args.channel.to_lowercase().as_str()], &BACKPORT_CHANNELS) {
+        return Err(anyhow::anyhow!(
+            "Parser error: unknown channel (allowed: {BACKPORT_CHANNELS:?})."
+        ));
+    }
+
+    let message_body = if contains_any(&[verb.as_str()], &BACKPORT_VERBS_APPROVE) {
+        BACKPORT_APPROVED
+    } else if contains_any(&[verb.as_str()], &BACKPORT_VERBS_DECLINE) {
+        BACKPORT_DECLINED
+    } else {
+        return Err(anyhow::anyhow!(
+            "Parser error: unknown verb (allowed: {BACKPORT_VERBS_APPROVE:?} or {BACKPORT_VERBS_DECLINE:?})"
+        ));
+    };
+
+    // TODO: factor out the Zulip "URL encoder" to make it practical to use
+    let zulip_send_req = crate::zulip::MessageApiRequest {
+        recipient: Recipient::Stream {
+            id: stream_id,
+            topic: &subject,
+        },
+        content: "",
+    };
+    let zulip_link = zulip_send_req.url(zulip_client);
+
+    let message_body = message_body
+        .replace("{args.channel}", args.channel.as_str())
+        .replace("{args.verb}", verb.as_str())
+        .replace("{zulip_link}", zulip_link.as_str());
+
+    tokio::spawn(async move {
+        let res = octo_client
+            .issues(repo_owner, repo_name)
+            .create_comment(args.pr_num, &message_body)
+            .await
+            .context("unable to post comment on #{args.pr_num}");
```

Comment:
This will be formatted as `#{args.pr_num}`. You can use e.g. `with_context(|| anyhow::anyhow("unable to post comment on #{}", args.pr_num))`.